activerecord/lib/active_record/associations/builder/singular_association.rb

```diff
@@ -19,6 +19,10 @@ def self.define_accessors(model, reflection)
         def reload_#{name}
           association(:#{name}).force_reload_reader
         end
+
+        def #{name}!
+          raise ActiveRecord::RecordNotFound unless association(:#{name}).force_reload_reader.present?
+        end
```

Comment:
There's probably a more elegant way to do this, but I wanted to explore the simplest possible solution to get my test passing.

Reply:
I also need to return the record if it's found.

Reply:
Addressed here:

- 99afe249d39a7fb3523a0232750e8e68ea8b6a39